compiler.go

```diff
@@ -77,6 +79,9 @@ func (c *Compiler) MustCompile(url string) *Schema {
 // Compile parses json-schema at given url returns, if successful,
 // a Schema object that can be used to match against json.
 func (c *Compiler) Compile(url string) (*Schema, error) {
+	if err := c.initStaticResources(); err != nil {
+		return nil, err
+	}
```

Comment:
maybe do this in `NewCompiler` instead
```go
func NewCompiler() *Compiler {
	c := &Compiler{
		Draft: latest,
		resources: make(map[string]*resource),
		Extensions: make(map[string]Extension),
	}

	for _, d := range []*Draft{Draft7, Draft6, Draft4} {
		if err := c.AddResource(d.url, strings.NewReader(d.data)); err != nil {
			panic("could not add draft "+d.url+": "+err.Error())
		}
	}
	return c
}
```
panic should be ok as the error is non-recoverable (if one should arise).

Reply:
Then you can theoretically also later overwrite it without the need to check on insert.

Reply:
i see, that makes sense